paging-runtime-composeui/src/commonMain/kotlin/app/cash/paging/LazyPagingItems.kt

```diff
@@ -0,0 +1,320 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+
+import androidx.paging.CombinedLoadStates
+import androidx.paging.DifferCallback
+import androidx.paging.ItemSnapshotList
+import androidx.paging.LOGGER
+import androidx.paging.LoadState
+import androidx.paging.LoadStates
+import androidx.paging.NullPaddedList
+import androidx.paging.PagingData
+import androidx.paging.PagingDataDiffer
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect // TODO("OG") Why does compiler say this is unused?
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.withContext
+
+
+/**
+ * The class responsible for accessing the data from a [Flow] of [PagingData].
+ * In order to obtain an instance of [LazyPagingItems] use the [collectAsLazyPagingItems] extension
+ * method of [Flow] with [PagingData].
+ * This instance can be used by the [items] and [itemsIndexed] methods inside [LazyListScope] to
+ * display data received from the [Flow] of [PagingData].
+ *
+ * @param T the type of value used by [PagingData].
+ */
+class LazyPagingItems<T : Any> internal constructor(
+  /**
+   * the [Flow] object which contains a stream of [PagingData] elements.
+   */
+  private val flow: Flow<PagingData<T>>
+) {
+
+  private val mainDispatcher = createMainDispatcher()
+//  private val mainDispatcher = Dispatchers.Main // TOSO
+
+  /**
+   * Contains the immutable [ItemSnapshotList] of currently presented items, including any
+   * placeholders if they are enabled.
+   * Note that similarly to [peek] accessing the items in a list will not trigger any loads.
+   * Use [get] to achieve such behavior.
+   */
+  var itemSnapshotList by mutableStateOf(
+    ItemSnapshotList<T>(0, 0, emptyList())
+  )
+    private set
+
+  /**
+   * The number of items which can be accessed.
+   */
+  val itemCount: Int get() = itemSnapshotList.size
+
+  private val differCallback: DifferCallback = object : DifferCallback {
+    override fun onChanged(position: Int, count: Int) {
+      if (count > 0) {
+        updateItemSnapshotList()
+      }
+    }
+
+    override fun onInserted(position: Int, count: Int) {
+      if (count > 0) {
+        updateItemSnapshotList()
+      }
+    }
+
+    override fun onRemoved(position: Int, count: Int) {
+      if (count > 0) {
+        updateItemSnapshotList()
+      }
+    }
+  }
+
+  // TODO("OG") Fix
+  // Cannot access class 'kotlin.coroutines.CoroutineContext'.
+  // Check your module classpath for missing or conflicting dependencies
+  private val pagingDataDiffer = object : PagingDataDiffer<T>(
+    differCallback = differCallback,
+    mainContext = mainDispatcher
```

Comment:
- [ ] First blocker - I'm getting crashes/errors when trying to run desktop app with error below. Not sure what I am missing as I think I have setup the right dependencies... 

```
  // Cannot access class 'kotlin.coroutines.CoroutineContext'.
  // Check your module classpath for missing or conflicting dependencies
```